### ISSUE TITLE
Publishing JupyterBook

### ISSUE THREAD

**Issue by tsnow03** [CryoInTheCloud/CryoCloudWebsite#14]
The preview and test checks finish successfully for building our JupyterBook, but the deploy action is failing. I've recently added a deploy key and used that to create our secrets.GH_PAT. This is the failure still occurring: 

https://github.com/CryoInTheCloud/CryoCloudWebsite/actions/runs/3294422660/jobs/5432924163#:~:text=fatal%3A%20unable%20to%20access%20%27https%3A//mines.edu%40github.com/CryoInTheCloud/cryointhecloud.github.io.git/%27%3A%20Could%20not%20resolve%20host%3A%20mines.edu%40github.com

Any ideas on the issue?

**Comment by weiji14** [comment]
I think the error is this line on https://github.com/CryoInTheCloud/CryoCloudWebsite/actions/runs/3294422660/jobs/5432924163#step:4:841

```
fatal: unable to access '[https://mines.edu@github.com/CryoInTheCloud/cryointhecloud.github.io.git/](https://mines%2Eedu@github.com/CryoInTheCloud/cryointhecloud.github.io.git/)': Could not resolve host: mines.edu@github.com
  [INFO] first deployment, create new branch gh-pages
  [INFO] The process '/usr/bin/git' failed with exit code 128
```

My guess is that there were some changes to GitHub Pages on 10 Aug 2022 (see https://github.blog/2022-08-10-github-pages-now-uses-actions-by-default/) that might have affected the GitHub Actions workflows from https://github.com/uwhackweek/jupyterbook-template (which were designed for hackweeks in the early half of 2022). Specifically, I think the current workflow is trying to publish to a `gh-pages` branch, but that might not be the case anymore?

Or it could just be some misconfigured credentials, and I'm overthinking things.

**Comment by weiji14** [comment]
Ah, I see at https://github.com/CryoInTheCloud/CryoCloudWebsite/settings/pages that we're using 'GitHub Actions' as the page source. Should we be using 'Deploy from branch' instead?
![image](https://user-images.githubusercontent.com/23487320/197209494-195f0db3-0d10-4568-88eb-e56509dae919.png)

**Comment by weiji14** [comment]
Reading https://github.com/peaceiris/actions-gh-pages/blob/e1448a73ca97cd0b1a6f894340080366c5e8509a/README.md#%EF%B8%8F-first-deployment-with-github_token, that exact `fatal: unable to access` error seems to be documented. It recommends:

> Please add the write permission to the [permissions.contents in a workflow/job](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions).

```
permissions:
  contents: write
```

But I'm not sure if it applies since we're using `GH_PAT` instead of `GITHUB_TOKEN`?

**Comment by tsnow03** [comment]
Ok I will work on this later today. Thank you for the ideas!

**Comment by tsnow03** [comment]
Created a deploy key in CryoCloudWebsite repository with `ssh-keygen -t ed25519 -C "my@email" -f gh-pages -N "" `. Added the private (gh-pages) key to a secret (ACTIONS_DEPLOY_KEY) on the Website, and added the public (gh-pages.pub) as a deploy key on the IO - in accordance with {https://github.com/peaceiris/actions-gh-pages/blob/e1448a73ca97cd0b1a6f894340080366c5e8509a/README.md#%EF%B8%8F-create-ssh-deploy-key}.

We did it. Closing.
